lib/livebook/session.ex

```diff
@@ -474,10 +475,11 @@ defmodule Livebook.Session do
 
   def handle_info({:DOWN, _, :process, pid, _}, state) do
     state =
-      if pid in state.data.client_pids do
-        handle_operation(state, {:client_leave, pid})
-      else
-        state
+      state.data.clients
+      |> Enum.find(&match?({_user_id, ^pid}, &1))
```

Comment:
I see a lot of Enum.find/2. What we typically do in these cases is to keep two maps: one from user_id to pid and another from pid to user_id.

Reply:
We now have `clients_map` from `pid` to `user_id` and `users_map` from `user_id` to `User.t()`, so it's much more convenient.